e3sm_diags/logger.py

```diff
@@ -37,12 +18,33 @@
 logging.getLogger().addHandler(console_handler)
 
 
-def custom_logger(name: str, propagate: bool = True) -> logging.Logger:
-    """Sets up a custom logger that is a child of the root logger.
+def _setup_root_logger():
+    """Configures the root logger.
+
+    This function sets up the root logger with a predefined format and log level.
+    It also enables capturing of warnings issued by the `warnings` module and
+    redirects them to the logging system.
+
+    Notes
+    -----
+    - The `force=True` parameter ensures that any existing logging configuration
+      is overridden.
+    - The file handler is added dynamically to the root logger later in the
+      ``Run`` class once the log file path is known.
+    """
+    logging.basicConfig(
+        format=LOG_FORMAT,
+        level=LOG_LEVEL,
+        force=True,
+    )
+
+    logging.captureWarnings(True)
 
-    This custom logger inherits the root logger's handlers.
 
+def _setup_child_logger(name: str, propagate: bool = True) -> logging.Logger:
```

Comment:
I renamed this function from `_setup_logger()` to `_setup_child_logger()` to make it clearer.